### PR TITLE
Add Binance.app

### PR DESCRIPTION
**Pull request by pgreze**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask audit --new-cask {{cask_file}}` worked successfully.
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

### CODE REVIEW

File: Casks/binance.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cask "binance" do
+  version :latest
+  sha256 :no_check
+
+  url "https://ftp.binance.com/electron-desktop/mac/production/binance.dmg"
+  name "Binance"
+  desc "Cryptocurrency exchange"
+  homepage "http://binance.com/"
+
+  app "Binance.app"
```
Comment (vitorgalvao) [inline Casks/binance.rb:10]: Please add a [`zap`](https://github.com/homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/zap.md).

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Comment by core-code** [comment]
i think there is an appcast here: https://ftp.binance.com/desktop/appcast.xml

**Comment by vitorgalvao** [comment]
> i think there is an appcast here: https://ftp.binance.com/desktop/appcast.xml

It’s behind. It shows `1.2.23` but the unversioned URL on this PR downloads `1.5.0`.

**Comment by core-code** [comment]
next try :->

`https://ftp.binance.com/electron-desktop/mac/production/latest-mac.yml`

File: Casks/binance.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+cask "binance" do
+  version :latest
+  sha256 :no_check
+
+  url "https://ftp.binance.com/electron-desktop/mac/production/binance.dmg"
```
Comment (vitorgalvao) [inline Casks/binance.rb:5]: Please add an [`appcast`](https://github.com/homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/appcast.md), the latest one pointed out by @core-code: https://ftp.binance.com/electron-desktop/mac/production/latest-mac.yml.

File: Casks/binance.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+cask "binance" do
+  version :latest
```
Comment (vitorgalvao) [inline Casks/binance.rb:2]: Set to `1.5.0`.

File: Casks/binance.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  sha256 :no_check
```
Comment (vitorgalvao) [inline Casks/binance.rb:3]: `sha256` also needs to be set. You can find what it is by running `shasum --algorithm 256` on the downloaded file.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/binance.rb (hunk 2 above)
Comment (pgreze) [reply Casks/binance.rb]: @vitorgalvao thx for your help with @core-code , I could update the Cask with the appcast url (even if I have no idea what's appcast 😆 )

File: Casks/binance.rb (hunk 3 above)
Comment (pgreze) [reply Casks/binance.rb]: I chose the 1.5.1, version returned by my current latest-mac.yaml fetch

File: Casks/binance.rb (hunk 4 above)
Comment (pgreze) [reply Casks/binance.rb]: Thank you for giving me the command, the appcast was using sha512 which was misleading me...

**Comment by pgreze** [comment]
@vitorgalvao let me reassign you now that I've updated the Cast file. My `reinstall` command worked as expected 👍

File: Casks/binance.rb (hunk 1 above)
Comment (vitorgalvao) [reply Casks/binance.rb:11]: @pgreze Still missing the `zap`.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/binance.rb (hunk 1 above)
Comment (pgreze) [reply Casks/binance.rb:11]: @vitorgalvao oh I missed this one.

I searched for created files by this app with "find . | grep -i binance" command and after an install + use of the app including a login, here are logs of the `zap` command:

```
$ brew cask zap binance
==> Implied "brew cask uninstall binance"
==> Backing App 'Binance.app' up to '/usr/local/Caskroom/binance/1.5.1/Binance.app'.
==> Removing App '/Applications/Binance.app'.
==> Dispatching zap stanza
==> Trashing files:
~/Library/Application Support/Binance
~/Library/Saved Application State/com.binance.BinanceDesktop.savedState
~/Library/Logs/Binance
==> Removing all staged versions of Cask 'binance'
```

File: Casks/binance.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  url "https://ftp.binance.com/electron-desktop/mac/production/binance-#{version}.dmg"
+  appcast "https://ftp.binance.com/electron-desktop/mac/production/latest-mac.yml"
+  name "Binance"
+  desc "Cryptocurrency exchange"
+  homepage "http://binance.com/"
+
+  app "Binance.app"
+
+  zap trash: [
+               '~/Library/Application Support/Binance',
```
Comment (vitorgalvao) [inline Casks/binance.rb:14]: These don’t have the correct indentations and have the wrong quote type. `brew cask style --fix`, as in the template, should fix it.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/binance.rb (hunk 5 above)
Comment (pgreze) [reply Casks/binance.rb]: @vitorgalvao argh, I followed the [given dropbox.rb example](https://github.com/Homebrew/homebrew-cask/blob/31cd96cc0e00dab1bff74d622e32d816bafd1f6f/Casks/dropbox.rb#L17-L35) given in [this page](https://github.com/homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/zap.md) but I guess it's referring a commit before this styling change.

I just pushed a commit fixing the styling issue.

Comment (pgreze) [reply Casks/binance.rb]: https://github.com/Homebrew/homebrew-cask/pull/91671

**Review by vitorgalvao — APPROVED** [review]

**Comment by vitorgalvao** [comment]
Looking good. Thank you.

**Comment by pgreze** [comment]
@vitorgalvao thank you for your support, it was my first PR in this project for an app I'm not maintaining, so I really appreciated your support 🙏

**Pull request merged**